### ISSUE TITLE
Design System Staging Review - Accessibility Feedback - Converted components: Memorable date, Select and Breadcrumbs

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#64284]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

-

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

### Memorable date

- **Must (probably a bug):** There's a keyboard trap when in the error state. Once you tab into a memorable date with an error, you can tab through those inputs but can never escape.
- **Must:** Error messages are announced on blur thanks to `role="alert"`, but if there's an error on the component and you're tabbing in from elsewhere, you won't hear the message when the component receives focus. Non-focusable elements inside the `fieldset` just get ignored by screen readers, unless you do some ugly ARIA (with mixed screen reader support) or you bundle them all into the `legend`. Since the hint text is already in the `legend`, I think it makes sense to put the error message in there too.

### Select

- **Must:** Hint text and error text aren't programmatically associated with the `select`. Recommend using `aria-describedby` to ensure screen readers will announce them.

### Breadcrumbs

- **Should:** This is a departure from the USWDS, but the last item in the breadcrumbs indicating the current page should be a link with `aria-current="page"` applied to the `a` element. That makes it consistent with the other elements in the `nav`, including being focusable as screen reader users tab through the elements. @rsmithadhoc's [comment on this](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1458#issuecomment-1671807757) has a good thought process to back up this decision, and it's supported by the [W3C's ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/breadcrumb/) and some examples from [Scott O'Hara](https://scottaohara.github.io/a11y_breadcrumbs/) and [Léonie Watson](https://tink.uk/using-the-aria-current-attribute/).
   
   In terms of styling, I would suggest we still maintain material honesty (unlike the W3C and Scott O'Hara examples): the current page element should still look like a link. It might make sense to style the link differently (bold? different color? different underline style?) but it should still be recognizably linky. Hopefully that's an improvement that can also be contributed back to the USWDS.
    
   Marking this as a should since the component as it is now is semantically correct and consistent with other sources. But I think it's a strong should!

**Comment by jeana-adhoc** [comment]
@caw310  This was identified as a [launch blocking issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/64658) today for the release of our 21-0972 form.

**Comment by powellkerry** [comment]
Memorable date first must is complete with this ticket and associated PR: https://github.com/department-of-veterans-affairs/va.gov-team/issues/64658

**Comment by tbaker1026** [comment]
@powellkerry Is this one done and merged to production?  Do we need to get code from you in order to fix it on our form?  Thank you

**Comment by caw310** [comment]
Work on this ticket is only for the Must haves. WIll write up a separate ticket for the should.
